python/python/bystro/beanstalkd/t/t_messages.py

```diff
@@ -0,0 +1,205 @@
+import unittest
+from enum import Enum
+
+from msgspec import Struct
+from msgspec.inspect import type_info, Field, IntType, NODEFAULT
+
+from bystro.beanstalkd.messages import (
+    SubmissionID,
+    BeanstalkJobID,
+    Event,
+    BaseMessage,
+    SubmittedJobMessage,
+    CompletedJobMessage,
+    FailedJobMessage,
+    InvalidJobMessage,
+    ProgressData,
+    ProgressMessage,
+)
+
+
+class T(Struct):
+    a: int
+    b: str
+
+
+class ImmutableT(Struct, frozen=True):
+    a: int
+    b: str
+
+
+class DefaultT(Struct, frozen=True):
+    a: int
+    b: str = "test"
+
+
+class InheritedT(T):
+    c: str
+
+
+class TestYourModule(unittest.TestCase):
+    def test_struct(self):
+        t = T(a=1, b="test")
+        self.assertEqual(t.a, 1)
+        self.assertEqual(t.b, "test")
+
+        t_immutable = ImmutableT(a=1, b="test")
+        with self.assertRaisesRegex(AttributeError, "immutable type: 'ImmutableT'"):
+            t_immutable.a = 2
+
+        t_default = DefaultT(a=1)
+        self.assertEqual(t_default.b, "test")
+
+        with self.assertRaisesRegex(TypeError, "Missing required argument 'c'"):
+            InheritedT(a=5, b="test")
+
+    def test_event_enum(self):
+        self.assertTrue(issubclass(Event, Enum))
+        self.assertEqual(Event.PROGRESS, "progress")
+        self.assertEqual(Event.FAILED, "failed")
+        self.assertEqual(Event.STARTED, "started")
+        self.assertEqual(Event.COMPLETED, "completed")
+
+    def test_base_message(self):
+        self.assertTrue(issubclass(BaseMessage, Struct))
+
+        t = BaseMessage(submissionID="test")
```

Comment:
In tests I've found it's often worth spelling things out a little more than one normally would in application code, because dealing with test code more often involves a context switch (you were doing something else far off in the codebase, but now suddenly you have to deal with a broken test here...).

Something like `base_message = BaseMessage(...)` could be a help to the reader, especially since a few lines above we were using `t` for variables of the superclass.